src/connector/src/parser/additional_columns.rs

```diff
@@ -87,75 +88,172 @@ fn kafka_compatible_column_vec() -> Vec<(&'static str, CompatibleAdditionalColum
                     column_desc: ColumnDesc::named_with_additional_column(
                         name,
                         id,
-                        DataType::Int64,
+                        DataType::Varchar,
                         AdditionalColumnType::Offset,
                     ),
                     is_hidden: false,
                 }
             }),
         ),
-        // Todo(tabVersion): add header column desc
-        // (
-        //     "header",
-        //     Box::new(|id: ColumnId, name: &str| -> ColumnCatalog {
-        //         ColumnCatalog {
-        //             column_desc: ColumnDesc::named(name, id, DataType::List(
-        //
-        //             )),
-        //             is_hidden: false,
-        //         }
-        //     }),
-        // ),
+        (
+            "header", // type: struct<key varchar, value bytea>[]
```

Comment:
I think `JSONB` is better for storing Kafka header because it can get (`->`) a value easily.

Somehow we should follow the same discussion at https://github.com/risingwavelabs/risingwave/issues/13387#issuecomment-1833403546

Reply:
The problem is jsonb type do not support bytes inside.

Reply:
Good point. 🤣 Thinking...

Reply:
We can support `array_filter()` https://github.com/risingwavelabs/rfcs/pull/69/files#diff-857a6f40f71644499fee9c269c260a570942420de9a0225b059508d02c1fe98bR127-R138 or `array_find` for it. if there is not too many fields in the array.
Or support `Map` datatype...